Sources/Group.php

```diff
@@ -2240,10 +2240,10 @@ public static function getAllWithPermissions(string|array $permissions, ?int $bo
 		}
 
 		// Maybe a mod needs to tweak the list of allowed groups on the fly?
-		IntegrationHook::call('integrate_groups_with_permissions', [&$groups, $permissions, $board]);
+		IntegrationHook::call('integrate_groups_with_permissions', [&$groups, $permissions, $board_or_profile]);
 
 		// Call the deprecated integrate_groups_allowed_to hook.
-		self::integrateGroupsAllowedTo($groups, $permissions, $board);
+		self::integrateGroupsAllowedTo($groups, $permissions, $board_or_profile);
```

Comment:
Hm. Yes, there is a problem here, but this solution won't suffice because `$board_or_profile` might be a profile. We need to make some more substantial changes to Group::getAllWithPermissions() to fix it. I will add some changes to this PR in order to do that.